system-parachains/asset-hubs/asset-hub-kusama/src/ah_migration/call_filter.rs

```diff
@@ -0,0 +1,217 @@
+		let (_before, during, _after) = call_allowed_status(call);
+		if !during {
+			log::warn!("Call bounced by the filter during the migration: {:?}", call);
+		}
+		during
+	}
+}
+
+/// Contains all calls that are enabled after the migration.
+pub struct CallsEnabledAfterMigration;
+impl Contains<<Runtime as frame_system::Config>::RuntimeCall> for CallsEnabledAfterMigration {
+	fn contains(call: &<Runtime as frame_system::Config>::RuntimeCall) -> bool {
+		let (_before, _during, after) = call_allowed_status(call);
+		if !after {
+			log::warn!("Call bounced by the filter after the migration: {:?}", call);
+		}
+		after
+	}
+}
+
+/// Return whether a call should be enabled before, during and/or after the migration.
+///
+/// Time line of the migration looks like this:
+///
+/// --------|-----------|--------->
+///       Start        End
+///
+/// We now define 3 periods:
+///
+/// 1. Before the migration: [0, Start)
+/// 2. During the migration: [Start, End]
+/// 3. After the migration: (End, ∞)
+///
+/// Visually:
+///
+/// ```text
+/// |--1---|
+///         |-----2-----|
+///                      |---3---->
+/// --------|-----------|--------->
+///       Start        End
+/// ```
+///
+/// This call returns a 3-tuple to indicate whether a call is enabled during these periods.
+pub fn call_allowed_status(
+	call: &<Runtime as frame_system::Config>::RuntimeCall,
+) -> (bool, bool, bool) {
+	use RuntimeCall::*;
+	const ON: bool = true;
+	const OFF: bool = false;
+	let before_migration = call_allowed_before_migration(call);
+
+	let during_migration = match call {
+		AhMigrator(..) => ON, // required for the migration, only permissioned calls
+		AhOps(..) => OFF,     // Not needed during the migration
+		AssetConversion(..) => ON, // no reason to disable it, just convenience
+		AssetRate(..) => OFF,
+		Assets(..) => ON,   // no reason to disable it, just convenience
+		Balances(..) => ON, // no reason to disable it, just convenience
+		Bounties(..) => OFF,
+		ChildBounties(..) => OFF,
+		Claims(..) => OFF,
+		CollatorSelection(..) => ON, // Why?
+		ConvictionVoting(..) => OFF,
+		CumulusXcm(..) => OFF, /* Empty call enum, see https://github.com/paritytech/polkadot-sdk/issues/8222 */
+		ForeignAssets(..) => ON, // no reason to disable it, just convenience
+		Indices(..) => OFF,
+		MultiBlockElection(..) => OFF,
+		MultiBlockElectionSigned(..) => OFF,
+		MultiBlockElectionUnsigned(..) => OFF,
+		MultiBlockElectionVerifier(..) => OFF,
+		MessageQueue(..) => ON, // contains non-permissioned service calls
+		Multisig(..) => OFF,
+		Nfts(..) => ON, // no reason to disable it, just convenience
+		NominationPools(..) => OFF,
+		ParachainInfo(parachain_info::Call::__Ignore { .. }) => ON, // Has no calls
+		ParachainSystem(..) => ON,                                  // Only inherent and root calls
+		PolkadotXcm(..) => ON,                                      /* no reason to disable it, */
```

Comment:
I have an issue with this. Based on my info, if this is ON, one can queue downward messages to `KRC` during the migration, at least annoying the migration. If we are to be safe, why not put it as off? maybe missing something. Some comment about the reasoning would be useful.

Same applies te KRC, if `XcmPallet` is enabled there. One could queue upward Teleport/Transact(Remarks) to exhaust the DMP.